### PR TITLE
Fix archiving of smarty results

### PR DESCRIPTION
**Pull request by BruceForstall**
Run the command to ZIP the output directory in the same
"Execute Batch File" step as the smarty. Otherwise, a failure
of smarty will cause Jenkins to not run the subsequent batch file
steps, including the ZIP step.

Also, preserve the smarty error code to use that as the batch
file step exit code.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet-bot test CI please

File: netci.groovy [hunk 1]
```diff
@@ -2608,11 +2611,16 @@ Constants.allScenarios.each { scenario ->
...
-                                batchFile(buildCommands)
+                                // ZIP up the smarty output, no matter what the smarty result.
+                                addCommand("powershell -NoProfile -Command \"Add-Type -Assembly 'System.IO.Compression.FileSystem'; [System.IO.Compression.ZipFile]::CreateFromDirectory('.\\bin\\tests\\${osGroup}.${architecture}.${configuration}\\Smarty.run.0', '.\\bin\\tests\\${osGroup}.${architecture}.${configuration}\\Smarty.run.0.zip')\"")
 
-                                // ZIP up the smarty output.
-                                batchFile("powershell -NoProfile -Command \"Add-Type -Assembly 'System.IO.Compression.FileSystem'; [System.IO.Compression.ZipFile]::CreateFromDirectory('.\\bin\\tests\\${osGroup}.${architecture}.${configuration}\\Smarty.run.0', '.\\bin\\tests\\${osGroup}.${architecture}.${configuration}\\Smarty.run.0.zip')\"")
+                                // Use the smarty errorlevel as the script errorlevel.
+                                addCommand("exit /b %__save_smarty_errorlevel%")
+
+                                batchFile(buildCommands)
```
Comment (jashook) [inline netci.groovy]: Make sure this does not create the batchfile twice

**Comment by BruceForstall** [comment]
@dotnet-bot test CI please

**Comment by BruceForstall** [comment]
ugh
```
16:33:58 ERROR: Error fetching remote repo 'origin'
16:33:58 
hudson.plugins.git.GitException: Failed to fetch from https://github.com/dotnet/dotnet-ci
16:33:58 	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:817)
16:33:58 	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1084)
16:33:58 	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1115)
16:33:58 	at org.jenkinsci.plugins.multiplescms.MultiSCM.checkout(MultiSCM.java:143)
16:33:58 	at hudson.scm.SCM.checkout(SCM.java:495)
16:33:58 	at hudson.model.AbstractProject.checkout(AbstractProject.java:1212)
16:33:58 	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:566)
16:33:58 	at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
16:33:58 	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:491)
16:33:58 	at hudson.model.Run.execute(Run.java:1737)
16:33:58 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
16:33:58 	at hudson.model.ResourceController.execute(ResourceController.java:97)
16:33:58 	at hudson.model.Executor.run(Executor.java:421)
```

**Comment by BruceForstall** [comment]
@dotnet-bot test CI please

**Pull request merged**